token-metadata/js/src/state.ts

```diff
@@ -1,18 +1,18 @@
 import { PublicKey } from '@solana/web3.js';
-import {
-    getArrayDecoder,
-    getArrayEncoder,
-    getBytesDecoder,
-    getBytesEncoder,
-    getStructDecoder,
-    getStructEncoder,
-    getTupleDecoder,
-    getTupleEncoder,
-} from '@solana/codecs-data-structures';
-import { getStringDecoder, getStringEncoder } from '@solana/codecs-strings';
+import { getArrayCodec, getBytesCodec, getStructCodec, getTupleCodec } from '@solana/codecs-data-structures';
+import { getStringCodec } from '@solana/codecs-strings';
 
 export const TOKEN_METADATA_DISCRIMINATOR = Buffer.from([112, 132, 90, 90, 11, 88, 157, 87]);
 
+export const tokenMetadataCodec = getStructCodec([
```

Comment:
Let's _not_ export this, since we are peppering in some custom logic for all-zero update authorities, etc. Someone might get confused if they try to use the codec in its raw form.